### ISSUE TITLE
pushing with `bump_aliases: true` + `version: ...` fails

### ISSUE THREAD

**Issue by aoldershaw** [concourse/registry-image-resource#271]
We're seeing this when trying to release a new version of the registry-image resource: https://ci.concourse-ci.org/teams/main/pipelines/resource/jobs/publish-patch/builds/1.1?vars.type=%22registry-image%22

```
selected worker: ci-worker-2
INFO[0000] pushing tag(s) 1.2.1, 1, 1.2                 
2021/03/24 17:48:44 existing blob: sha256:ba3557a56b150f9b813f9d02274d62914fd8fce120dd374d9ee17b87cf1d277d
2021/03/24 17:48:44 existing blob: sha256:aa1667d1faffcfbc397d36fb90951cd450feedb0c870460c5af016c6a1757604
2021/03/24 17:48:44 existing blob: sha256:3c60fcb2402aafbbc4514277658c0a4e722760042254a69b90ed756b9ec0b5b1
2021/03/24 17:48:44 existing blob:  
sha256:344f8585e0c1336b7a8209c662609435a0970d810b7127360b71230dfb159665
2021/03/24 17:48:44 existing blob: sha256:6d11b0c5c0a9ae97dabbc0b94663ad837583596cf3ee3e86ad53664834beca17
ERRO[0001] pushing image failed: pushing tag(s): Put "https://index.docker.io/v2/concourse/registry-image-resource/manifests/1.2": http: ContentLength=917 with Body length 0 
```

We were able to reproduce locally using the following pipeline:

```yaml
resources:
- name: busybox
  type: registry-image
  source: {repository: busybox}

- name: image
  type: registry-image
  source:
    repository: concoursetest/test
    username: ...
    password: ...

jobs:
- name: push
  plan:
  - get: busybox
    params: {format: oci}
  - put: image
    params:
      image: busybox/image.tar
      version: 1.1.0
      bump_aliases: true
```

**Comment by aoldershaw** [comment]
Note: this isn't (just) a problem with pushing multiple tags - using `additional_tags` instead of `bump_aliases: true` + `version: ...` did not produce this error (and pushed successfully)

**Comment by aoldershaw** [comment]
seems to have been fixed by #270 as well 🤷 added an integration test in #271
